tests/bugs/shard/issue-1384.t

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+. $(dirname $0)/../../include.rc
+. $(dirname $0)/../../volume.rc
+
+cleanup;
+
+TEST glusterd
+TEST pidof glusterd
+TEST $CLI volume create $V0 replica 3 $H0:$B0/${V0}{0,1,3}
+TEST $CLI volume set $V0 features.shard on
+TEST $CLI volume set $V0 features.shard-block-size 4MB
+TEST $CLI volume start $V0
+
+TEST $GFS --volfile-id=$V0 --volfile-server=$H0 $M0
+
+TEST $CLI volume set $V0 md-cache-timeout 10
+
+# Write data into a file such that its size crosses shard-block-size
+TEST dd if=/dev/zero of=$M0/foo bs=1048576 count=10 oflag=direct
+
+# Size of the file should be the aggregated size, not the shard-block-size
+EXPECT "10485760" echo `ls -la $M0 | grep foo | awk '{print $5}'`
+
+TEST $CLI volume set $V0 performance.read-ahead on
+TEST $CLI volume set $V0 performance.parallel-readdir on
+
+sleep 5
```

Comment:
You need to trigger a `readdir` after enabling readdir-ahead to force it to cache the data. Umounting and remounting the volume before that will also make sure that other caches are not interfering here.

Also, is it necessary this `sleep` ?

Reply:
> You need to trigger a readdir after enabling readdir-ahead to force it to cache the data.
Yep, triggering ls in line #31

> Also, is it necessary this sleep ?
It takes sometime to readdir-ahead to update the cache. so kept sleep there

> Umounting and remounting the volume before that will also make sure that other caches are not interfering here.
Should I umount and mount after enabling `readdir-ahead` and remove `ls` ?

Reply:
> Yep, triggering ls in line **#31**

Yes, you are right.

> It takes sometime to readdir-ahead to update the cache. so kept sleep there

Without any command sent, caches won't be updated. What's the difference between having the sleep and not having it ?

If the problem is that the volume is not yet ready after the change, then you should check that explicitly instead of using a sleep that could spuriously fail.

> Should I umount and mount after enabling readdir-ahead and remove ls ?

The `ls` shouldn't be removed. It's what we want to test, right ? but, yes, I would umount and remount to clear any potential caches that could hide the effects of the patch.

Reply:
Sure. I Will updated the .t with the suggested changes